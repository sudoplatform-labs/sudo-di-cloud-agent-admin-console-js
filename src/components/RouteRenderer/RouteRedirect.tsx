import React from 'react';
import { Redirect, generatePath } from 'react-router';

interface Props {
  routePath: string;
}

export const RouteRedirect: React.FC<Props> = (props) => {
  return <Redirect to={generatePath(props.routePath)} />;
};

export function createRouteRedirect(routePath: string): React.FC {
  const BoundRouteRedirect: React.FC = () => (
    <RouteRedirect routePath={routePath} />
  );

  return BoundRouteRedirect;
}
