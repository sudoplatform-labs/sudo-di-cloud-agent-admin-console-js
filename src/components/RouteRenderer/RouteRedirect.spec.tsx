import { shallow } from 'enzyme';
import * as React from 'react';
import { RouteRedirect, createRouteRedirect } from './RouteRedirect';

describe('components/RouteRenderer/RouteRedirect', () => {
  describe('RouteRedirect', () => {
    it('should render a redirect', () => {
      const wrapper = shallow(<RouteRedirect routePath="/path" />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('createRouteRedirect()', () => {
    it('should create a component that renders a RouteRedirect for the given route path', () => {
      const Component = createRouteRedirect('/my-path');
      const wrapper = shallow(<Component />);
      expect(wrapper.find(RouteRedirect).prop('routePath')).toBe('/my-path');
    });
  });
});
