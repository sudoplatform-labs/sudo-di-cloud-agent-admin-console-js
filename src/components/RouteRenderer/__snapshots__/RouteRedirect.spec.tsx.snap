// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RouteRenderer/RouteRedirect RouteRedirect should render a redirect 1`] = `
<Redirect
  to="/path"
/>
`;
